lib/gateway/pipelines.js

```diff
@@ -199,14 +192,35 @@ function normalizeGatewayConfig (config) {
   return true;
 }
 
-function validatePipelinePolicies (policies, config) {
-  const policiesConfig = config.gatewayConfig.policies || [];
-
+function validatePipelinePolicies (policies, avaiablePolicies) {
   policies.forEach(policyObj => {
     const policyNames = Object.keys(policyObj);
 
-    if (policiesConfig.indexOf(policyNames[0]) === -1) {
+    if (avaiablePolicies.indexOf(policyNames[0]) === -1) {
       throw new Error(`${policyNames[0]} policy not declared`);
     }
   });
 }
+
+const generatePipelineHandler = (path, pipeline, route) => {
```

Comment:
may be object  `{path, pipeline, route}`  ?

Reply:
Done in 31b8497